File: src/plugins/hello.lua
-- ijevim Lua plugin example
-- API: addCommand(name, fn), on(event, fn), log(msg)

return {
  name = "hello-lua",
  version = "0.1.0",

  setup = function(api)
    api.addCommand("hello-lua", function()
      api.log("Hello from Lua plugin!")
    end)

    api.on("Ready", function()
      api.log("hello-lua: editor is ready")
    end)
  end
}

File: src/plugins/hello.js
// ijevim JavaScript plugin example
// API: ijevim.addCommand(name, func), ijevim.on(event, handler), ijevim.log(msg)

ijevim.addCommand("hello-js", function() {
    ijevim.log("Hello JS command called!");
});

ijevim.on("ready", function() {
    ijevim.log("JS plugin: editor is ready!");
});

ijevim.log("Hello from JS plugin!");
